File: app/src/main/java/com/example/jarsical/fragments/home/songAdapter.java
package com.example.jarsical.fragments.home;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.jarsical.PlaySongActivity;
import com.example.jarsical.R;
import com.example.jarsical.Song;
import com.example.jarsical.SongCollection;
import com.squareup.picasso.Picasso;

public class songAdapter extends RecyclerView.Adapter<songAdapter.ViewHolder> {
    public songAdapter(Song[] songs) {
        this.songs = songs;
    }

    Song[] songs;

    @NonNull
    @Override
    public songAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.custom_row,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull songAdapter.ViewHolder holder, int position) {
        Song song = songs[position];
        holder.songTitle.setText(song.getTitle());
        holder.songArtist.setText(song.getArtists());
        Picasso.get().load(song.getArtLink()).fit().placeholder(R.drawable.ic_baseline_error_24).into(holder.imageButton);// loads images into imageButton

        holder.imageButton.setContentDescription(song.getId());
        holder.favButton.setContentDescription(song.getId());

        holder.imageButton.setOnClickListener(v -> {
            Intent intent = new Intent(v.getContext(), PlaySongActivity.class);
            intent.putExtra("index",position);
            v.getContext().startActivity(intent);
        });

    }

    @Override
    public int getItemCount() {
        return songs.length;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView songTitle, songArtist;
        ImageButton imageButton;
        Button favButton;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            songTitle = itemView.findViewById(R.id.txtSongName);
            songArtist = itemView.findViewById(R.id.txtSongArtist);
            imageButton = itemView.findViewById(R.id.imageButton);
            favButton = itemView.findViewById(R.id.buttonTest);

        }
    }
}

File: app/src/main/java/com/example/jarsical/fragments/list/songAdapter.kt
package com.example.jarsical.fragments.list

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.jarsical.R
import com.example.jarsical.R.layout.custom_row2
import com.example.jarsical.Song
import com.example.jarsical.SongCollection
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.custom_row.view.*

class songAdapter: RecyclerView.Adapter<songAdapter.MyViewHolder>() {

    var songCollection = SongCollection()

    class MyViewHolder(itemView: View): RecyclerView.ViewHolder(itemView){

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(parent.context).inflate(custom_row2, parent, false))
    }

    override fun getItemCount(): Int {
        return songCollection.songs.size
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val currentItem = songCollection.songs[position]

        holder.itemView.txtSongName.text = currentItem.title
        holder.itemView.txtSongArtist.text = currentItem.artists
        holder.itemView.txtSongLength.text = currentItem.songLength.toString()
        //picasso allows us to use cover art without downloading
        Picasso.get().load(currentItem.artLink).placeholder(R.drawable.ic_baseline_error_24).into(holder.itemView.imageButton)// loads images into imageButton
        //sets content description
        holder.itemView.buttonTest.contentDescription = currentItem.id

        /*Log.d("test", holder.itemView.buttonTest.contentDescription.toString())*/


        val test = currentItem.id
        /*Log.d("onBind",test)*/

    }
    fun setData(song: Array<Song>){
        this.songCollection.songs = song
        notifyDataSetChanged()
    }
}